internal/convert/registryv1.go

```diff
@@ -279,13 +276,19 @@ func Convert(in RegistryV1, installNamespace string, targetNamespaces []string)
 
 	for _, permission := range permissions {
 		saName := saNameOrDefault(permission.ServiceAccountName)
-		name := generateName(fmt.Sprintf("%s-%s", in.CSV.Name, saName), []interface{}{in.CSV.Name, permission})
+		name, err := generateName(fmt.Sprintf("%s-%s", in.CSV.Name, saName), []interface{}{in.CSV.Name, permission})
```

Comment:
I didn't change this downstream for simplicity, but you don't need to include `in.CSV.Name` in the hash if it's in the prefix.

Also, use an anonymous struct here to make it JSON-serializable (see my changes in v0).

Reply:
Also, turns out getting rid of the CSV name means I can just pass `permission` directly without the interface slice wrapper.